core/navigation.js

```diff
@@ -387,6 +387,103 @@ Blockly.Navigation.resetFlyout = function(shouldHide) {
 /** Modify **/
 /************/
 
+/**
+ * Handle the modifier key (currently I for Insert).
+ * @return {boolean} True if the key was handled; false if something went wrong.
+ * @package
+ */
+Blockly.Navigation.modify = function() {
+  var markerNode = Blockly.Navigation.marker_.getCurNode();
+  var cursorNode = Blockly.Navigation.cursor_.getCurNode();
+
+  if (!markerNode) {
+    Blockly.Navigation.warn('Cannot insert with no marked node.');
+    return false;
+  }
+
+  if (!cursorNode) {
+    Blockly.Navigation.warn('Cannot insert with no cursor node.');
+    return false;
+  }
+
+  if (cursorNode.getType() == Blockly.ASTNode.types.FIELD) {
```

Comment:
Should we also have this check for a field marker node?

Reply:
Per previous discussion, we aren't going to let the marker be on a field.  But I'm not sure where we're going to block that, so yeah--it's probably worth having this check and fail if the marker is on a field.  I'll add it.